File: CS/HowToChangeFilterByTextAction.Module/MyFilterController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.SystemModule;
using DevExpress.Data.Filtering;
using DevExpress.ExpressApp.Filtering;
using DevExpress.ExpressApp.NodeWrappers;
using DevExpress.ExpressApp.DC;


namespace HowToChangeFilterByTextAction.Module {
   public partial class MyFilterController : FilterController{
      protected override void FullTextSearch(ParametrizedActionExecuteEventArgs args) {
         CriteriaOperator criteriaOperator = FullTextSearchCriteriaBuilder.BuildSimpleSearchCriteria(
            ObjectSpace.TypesInfo.FindTypeInfo(View.ObjectType),
            GetProperties(), (string)args.ParameterCurrentValue, GroupOperatorType.And);
         View.CollectionSource.Criteria[FullTextSearchCriteriaBuilder.CriteriaName] = criteriaOperator;
      }
      //Return the properties to be tested
      private ICollection<string> GetProperties() {
         List<string> result = new List<string>();
         foreach (ColumnInfoNodeWrapper column in View.Model.Columns.Items) {
            foreach (IMemberInfo member in ObjectSpace.TypesInfo.FindTypeInfo(View.ObjectType).Members) {
               if ((column.PropertyName == member.Name) && (member.IsPersistent) && (column.VisibleIndex != -1)) {
                  result.Add(column.PropertyName);
               }
            }
         }
         return result;
      }
   }
}

File: VB/HowToChangeFilterByTextAction.Module/MyFilterController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base

Imports DevExpress.ExpressApp.SystemModule

Namespace HowToChangeFilterByTextAction.Module
	Partial Public Class MyFilterController
		Inherits ViewController
		Public Sub New()
			InitializeComponent()
			RegisterActions(components)
			Me.TargetObjectType = GetType(DevExpress.Persistent.BaseImpl.Person)
		End Sub

		Private Sub MyFilterController_Activated(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Activated
			Dim standardFilterController As FilterController = Frame.GetController(Of FilterController)()
			If standardFilterController IsNot Nothing Then
				AddHandler standardFilterController.CustomGetFullTextSearchProperties, AddressOf standardFilterController_CustomGetFullTextSearchProperties
			End If
		End Sub

		Private Sub standardFilterController_CustomGetFullTextSearchProperties(ByVal sender As Object, ByVal e As CustomGetFullTextSearchPropertiesEventArgs)
			For Each [property] As String In GetFullTextSearchProperties()
				e.Properties.Add([property])
			Next [property]
			e.Handled = True
		End Sub
		Private Function GetFullTextSearchProperties() As List(Of String)
			Dim searchProperties As New List(Of String)()
			searchProperties.Add("LastName")
			Return searchProperties
		End Function
	End Class
End Namespace
